src/threading.c

```diff
@@ -771,6 +771,10 @@ void jl_init_threading(void)
     }
     int16_t ngcthreads = jl_n_markthreads + jl_n_sweepthreads;
 
+#ifdef MMTK_GC
+    ngcthreads = 0;
+#endif
```

Comment:
Now that you introduced `jl_active_gc_impl`, could we use that function here to set `ngcthreads` to 0, instead of the `#ifdef`?

I feel this is a bit more abstract, and I'd prefer to avoid putting a bunch of `#ifdef MMTK_GC` in the codebase.